### ISSUE TITLE
The rc shell is not supported

### ISSUE THREAD

**Issue by dertuxmalwieder** [purcell/exec-path-from-shell#114]
The README claims:

> Further, if you use a non-POSIX-standard shell (...), your shell will be asked to execute `sh` as a subshell in order to print out the variables in a format which can be reliably parsed. sh must be a POSIX-compliant shell in this case.

My `init.el` says:

```lisp
(when (eq system-type 'darwin)
  (elpaca exec-path-from-shell
    (setq exec-path-from-shell-check-startup-files nil)
    (exec-path-from-shell-initialize)))
```

I can confirm that `tcsh` works just well. Anyway, my login (= default) shell on this computer is [the `rc` shell](https://github.com/rakitzis/rc), arguably a *UNIX-y shell* (being the default shell of the last Research Unix version 10). Emacs does not like that:

> Warning (emacs): Config Error exec-path-from-shell: (error "Expected printf output from shell, but got: \"rc: syntax error
\"")

Suggestion: Add the `rc` executable to the list of non-standard shells:

```
; echo $SHELL
/Users/tux0r/bin/rc
```

**Comment by purcell** [comment]
Thanks, I'd potentially be open to a patch, and there's another minority shell (nushell) in a similar situation — see #109. Essentially, I guess those shells use different quoting/escaping rules, so the `printf` command needs to be assembled differently. Ideally we'd figure out a way to generalise this a little.

**Comment by dertuxmalwieder** [comment]
Depending on the implementation of rc, quoting is only one of the things which are different. Having used the Nushell for a while, I doubt that a common denominator will be rather hard to find, to be honest. It would probably still require a separate handling for special shells and their needs.

**Comment by purcell** [comment]
Yeah, I'm imagining maybe an overriding argument quoting function that's specified in an alist.

**Comment by dertuxmalwieder** [comment]
The recent addition of support for Nushell has made the code even harder for me to understand, I'm afraid. I had just tried a shot in the dark by trying to get anything sensible out of `rc` (essentially: `rc -c 'echo $path'`, so nothing too uncommon, minus the lower case) in `*scratch*`. The command works, but now I have no idea how to get the result into an Emacs list. 😂

**Comment by purcell** [comment]
The reason `rc` doesn't work turns out mostly because the existing code uses the `${SOMEVAR-somedefaultvalue}` syntax. Fixing it properly will involve adding some extension points to the code: having added `nushell` now, I have a better idea of how that would look, so I'll try to have a tinker with it sometime soon.

**Comment by dertuxmalwieder** [comment]
If you need any second pair of eyes, I’m here. :-)
My hands seem to be useless this time.

**Comment by dertuxmalwieder** [comment]
Hmm. Printing as many environment variables as you wish is easy (`echo $home $path $something...`), but my approach to check undefined values would be to use `rc`'s built-in counter:

```
; echo $#path
12                    # $path has 12 values

; echo $#home
1                     # unsurprisingly, $home has 1 value

; echo $#undefined
0                     # $undefined has 0 values ;-)
```
